diff -ruN occu-3.45.7.orig/firmware/HM-MOD-UART/fwmap occu-3.45.7.patched/firmware/HM-MOD-UART/fwmap
--- occu-3.45.7.orig/firmware/HM-MOD-UART/fwmap
+++ occu-3.45.7.patched/firmware/HM-MOD-UART/fwmap
@@ -1,7 +1,7 @@
 #Type           Filename                Version
 
 #Coprozessor
-CCU2                    coprocessor_update.eq3                         	1.4.1		# CoProzessor HM-MOD-UART
+#CCU2                    coprocessor_update.eq3                         	1.4.1		# CoProzessor HM-MOD-UART
 
-#CCU2			dualcopro_si1002_update_blhm.eq3		2.8.6		# Dual CoProzessor HM-MOD-UART
+CCU2			dualcopro_si1002_update_blhm.eq3		2.8.6		# Dual CoProzessor HM-MOD-UART
 
